x-pack/plugins/remote_clusters/public/shim.ts

```diff
@@ -21,19 +20,14 @@ export function createShim() {
     coreStart: {
       chrome,
       i18n,
-      notifications: {
-        ...notifications,
-        fatalError,
-      },
+      notifications,
+      fatalError,
```

Comment:
Just realized I was wrong and fatalError is already exposed, but right now it's only on CoreSetup. I'll put up a PR for exposing it in CoreStart as well, but it should be the exact same interface as CoreSetup.

Instead of getting this from `ui/notify` you should be able to get it from `npSetup.core.fatalErrors`. This is important because the interface in `ui/notify` is slightly different than the one exposed by Core (but should be a very easy refactor).

Reply:
Great, thanks!

Reply:
Been thinking about this more, and I'm not sure it makes sense to support this in the new platform. In the new platform:

- If a plugin fails to setup or start, they can throw exceptions during their `setup` and/or `start` method and Core will automatically display that error in the full-page error UI.
- If an application fails to mount or throws some error in the UI, that app should handle it with an error boundary React component to display the error without taking down the entire frontend.

In the case of your migration, it may make sense to use `npSetup.core.fatalError` in a `catch` block around `start` in your shim and replace any other usages with a React error boundary component that also calls `npSetup.core.fatalError` for now.

Thoughts?

Reply:
> If an application fails to mount or throws some error in the UI, that app should handle it with an error boundary React component to display the error without taking down the entire frontend.

I agree with this principle. In terms of the ES UI apps, `fatalError` is only used in a dozen places so migrating off it won't be difficult. I wonder if the Platform team can ease the migration process by wrapping the mountpoint in an `EuiErrorBoundary` or framework-agnostic equivalent (maybe even a `failApp` method that can be used in lieu of `fatalError`). The idea being that an app developer has a mechanism for "bailing out" with a descriptive error message if something absolutely unexpected occurred, without needing to implement this themselves. Using this mechanism would display an error in place of the app, without taking down all of Kibana. This will also ensure that there's a visually consistent way of presenting this information so each app doesn't look different to the user when it fails in this manner.

Reply:
I agree, consistency here is ideal. I'll add that as an additional to the ApplicationService project. In the meantime, I think it's fine to go ahead and keep using `fatalError` until we have another solution.